app/src/main/java/dev/devdigi/music/connection/ServerConnection.kt

```
@@ -0,0 +1,116 @@
+                (scheme != "https" && !isPermittedLocalHttp) ||
+                uri.userInfo != null ||
+                uri.query != null ||
+                uri.fragment != null ||
+                (!isPermittedLocalHttp && hostWithoutTerminalDots == "localhost") ||
+                (!isPermittedLocalHttp && hostWithoutTerminalDots.endsWith(".localhost")) ||
+                hostWithoutTerminalDots.endsWith(".local") ||
+                host.startsWith('[') ||
+                host.all(Char::isDigit) ||
+                (uri.port != -1 && uri.port !in 1..65535) ||
+                (!isPermittedLocalHttp && isProhibitedIpv4(host)) ||
+                hasAmbiguousPath(uri.rawPath)
```

detekt (1.23.8)

[warning] This condition is too complex (18). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)